internal/controllers/device_config_controller_test.go

```diff
@@ -0,0 +1,216 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/volatiletech/sqlboiler/v4/boil"
+
+	"github.com/DIMO-Network/vehicle-signal-decoding/internal/config"
+	"github.com/DIMO-Network/vehicle-signal-decoding/internal/infrastructure/db/models"
+	"github.com/DIMO-Network/vehicle-signal-decoding/internal/test"
+	"github.com/gofiber/fiber/v2"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+)
+
+const migrationsDirRelPath = "../infrastructure/db/migrations"
+
+func TestGetPIDsByTemplate(t *testing.T) {
+
+	// Arrange: db and route setup
+	logger := zerolog.New(os.Stdout).With().
+		Timestamp().
+		Str("app", "vehicle-signal-decoding").
+		Logger()
+
+	ctx := context.Background()
+
+	//Spin up test database in a Docker container
+	pdb, container := test.StartContainerDatabase(ctx, t, migrationsDirRelPath)
+	defer func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	template := models.Template{
+		TemplateName: "exampleTemplate",
+		// etc
+	}
+	err := template.Insert(context.Background(), pdb.DBS().Writer, boil.Infer())
+	assert.NoError(t, err)
+
+	pc := models.PidConfig{
+		ID:              1,
+		TemplateName:    "exampleTemplate",
+		Header:          []byte("7E8"),
+		Mode:            []byte("01"),
+		Pid:             []byte("05"),
+		Formula:         "A*5",
+		IntervalSeconds: 60,
+		Version:         "1.0",
+	}
+
+	errr := pc.Insert(context.Background(), pdb.DBS().Writer, boil.Infer())
+	assert.NoError(t, errr)
+
+	c := NewDeviceConfigController(&config.Settings{Port: "3000"}, &logger, pdb.DBS().Reader.DB)
+	app := fiber.New()
+	app.Get("/device-config/pid/:template_name", c.GetPIDsByTemplate)
```

Comment:
let's have these routes match the production ones you have in api.go, will help later on for consistency

Reply:
done!